packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -0,0 +1,184 @@
+  AuthenticationType,
+  HasNecessaryCoveoPrivileges,
+  IsAuthenticated,
+  Preconditions,
+} from '@coveo/cli-commons/preconditions';
+import {HostedPage, New} from '@coveo/platform-client';
+import {createSearchPagesPrivilege} from '@coveo/cli-commons/preconditions/platformPrivilege';
+import {Flags} from '@oclif/core';
+import {readJsonSync, ensureDirSync, readFileSync} from 'fs-extra';
+import {DeployConfigError} from '../../lib/errors/deployErrors';
+import {join} from 'path';
+
+interface FileInput {
+  path: string;
+}
+
+interface JavaScriptFileInput extends FileInput {
+  isModule: boolean;
+}
+
+interface DeployConfig {
+  name: string;
+  dir: string;
+  htmlEntryFile: FileInput;
+  javascriptEntryFiles?: JavaScriptFileInput[];
+  javascriptUrls?: JavaScriptFileInput[];
+  cssEntryFiles?: FileInput[];
+  cssUrls?: FileInput[];
+}
+
+const FileInputSchema: Schema = {
+  type: 'object',
+  required: ['path'],
+  properties: {
+    path: {type: 'string'},
+  },
+};
+
+const JavaScriptFileInputSchema: Schema = {
+  type: 'object',
+  required: ['path', 'isModule'],
+  properties: {
+    path: {type: 'string'},
+    patisModuleh: {type: 'boolean'},
+  },
+};
+
+const DeployConfigSchema: Schema = {
+  type: 'object',
+  required: ['name', 'dir', 'htmlEntryFile'],
+  properties: {
+    name: {type: 'string'},
+    dir: {type: 'string'},
+    htmlEntryFile: FileInputSchema,
+    javascriptEntryFiles: {type: 'array', items: JavaScriptFileInputSchema},
+    javascriptUrls: {type: 'array', items: JavaScriptFileInputSchema},
+    cssEntryFiles: {type: 'array', items: FileInputSchema},
+    cssUrls: {type: 'array', items: FileInputSchema},
+  },
+};
+
+export default class Deploy extends CLICommand {
+  public static hidden = true; /* TODO: uncomment */
+  public static flags = {
+    pageId: Flags.string({
+      char: 'p',
+      description: 'The existing ID of the Hosted Page to be updated.',
+      helpValue: '7944ff4a-9943-4999-a3f6-3e81a7f6fb0a',
+      required: false,
+    }),
+    config: Flags.string({
+      char: 'c',
+      description: 'The path to the deploy JSON configuration.',
+      helpValue: 'coveo.deploy.json',
+      default: 'coveo.deploy.json',
+      required: false,
+    }),
+  };
+
+  @Trackable()
+  @Preconditions(
+    IsAuthenticated([AuthenticationType.OAuth]),
+    HasNecessaryCoveoPrivileges(createSearchPagesPrivilege)
+  )
+  public async run() {
+    const {flags} = await this.parse(Deploy);
+    const deployConfigPath = flags.config;
+    const deployConfig: DeployConfig =
+      readJsonSync(deployConfigPath, {throws: false}) || {};
+    this.validateDeployConfigJson(deployConfigPath, deployConfig);
+    const hostedPage = this.buildHostedPage(deployConfig);
+
+    if (flags.pageId) {
+      this.updatePage({...hostedPage, id: flags.pageId});
+      return;
+    }
+
+    try {
+      await this.createPage(hostedPage);
+    } catch (error) {
+      /**
+       * TODO: handle Platform Client 400 error e.g.:
```

Comment:
How should I standardize this one, it wasn't handled https://github.com/coveo/cli/pull/930

Reply:
I reckon you can prolly let it go (no need for try-catch).
The catch-finally clauses at the CLICommand level should handle it for you
(@y-lakhdar amarite?)

Reply:
Yeah, no need for that. the CLI base command will handle the error.
If you need the handle the error in a specific way, you can use the `catch` public method. If that's the case, don't forget to call the `super.catch()` once your done.
Example:
```
 public async catch(err?: Error & {exitCode?: number}) {
    // custom stuff
    return super.catch(err);
  }
```